File: PHP/src/NorwegianBlueParrot.php
<?php
namespace Parrot;

class NorwegianBlueParrot extends Parrot {

  /** @var double */
  private $voltage;
  /** @var boolean */
  private $isNailed;

  public function __construct($voltage, $isNailed) {
    $this->voltage = $voltage;
    $this->isNailed = $isNailed;
  }

  public function getSpeed() {
    return $this->isNailed ? 0 : $this->getBaseSpeedWithVoltage();
  }

  private function getBaseSpeedWithVoltage() {
    return min(24.0, $this->voltage * $this->getBaseSpeed());
  }

}

File: Java-PerfKata/src/test/java/perfkata/parrot/a20171207/NorwegianBlueParrot.java
package perfkata.parrot.a20171207;

public class NorwegianBlueParrot extends Parrot {
    private final double voltage;
    private final boolean isNailed;

    public NorwegianBlueParrot(double voltage, boolean isNailed) {
        super(null, 0, 0, false);
        this.voltage = voltage;
        this.isNailed = isNailed;
    }

    @Override
    public double getSpeed() {
        return (isNailed) ? 0 : getBaseSpeedForVoltage();
    }

    private double getBaseSpeedForVoltage() {
        return Math.min(24.0, voltage * getBaseSpeed());
    }

}
